src/xdg.rs

```diff
@@ -74,25 +78,42 @@ impl DesktopEntries {
                     Err(_) => continue,
                 };
 
+                // Ignore all groups other than the `Desktop Entry` one.
+                //
+                // Since `Desktop Entry` must be the first group, we just stop at the next group
+                // header.
+                let lines = desktop_file.lines().take_while(|line| {
+                    line.trim_end() == "[Desktop Entry]" || !line.starts_with('[')
+                });
+
                 let mut icon = None;
                 let mut exec = None;
                 let mut name = None;
 
-                for line in desktop_file.lines() {
-                    if let Some(value) = line.strip_prefix("Name=") {
-                        name = Some(value.to_owned());
-                    } else if let Some(value) = line.strip_prefix("Icon=") {
-                        icon = desktop_entries.loader.load(value, icon_size).ok();
-                    } else if let Some(value) = line.strip_prefix("Exec=") {
-                        // Remove %f/%F/%u/%U/%k variables.
-                        let filtered = value
-                            .split(' ')
-                            .filter(|arg| !matches!(*arg, "%f" | "%F" | "%u" | "%U" | "%k"));
-                        exec = Some(filtered.collect::<Vec<_>>().join(" "));
-                    }
+                // Find name, icon, and executable for the desktop entry.
+                for line in lines {
+                    // Get K/V pairs, allowing for whitespace around the assignment operator.
+                    let (key, value) = match line.split_once('=') {
+                        Some((key, value)) => (key.trim_end(), value.trim_start()),
+                        None => continue,
+                    };
 
-                    if icon.is_some() && exec.is_some() && name.is_some() {
-                        break;
+                    match key {
+                        "Name" => name = Some(value.to_owned()),
+                        "Icon" => icon = desktop_entries.loader.load(value, icon_size).ok(),
+                        "Exec" => {
+                            // Remove %f/%F/%u/%U/%k variables.
+                            let filtered = value
+                                .split(' ')
+                                .filter(|arg| !matches!(*arg, "%f" | "%F" | "%u" | "%U" | "%k"));
+                            exec = Some(filtered.collect::<Vec<_>>().join(" "));
+                        },
+                        // Ignore explicitly hidden entries.
+                        "NoDisplay" if value.trim() == "true" => {
+                            exec = None;
+                            break;
+                        },
```

Comment:
This is kinda hacky, but should have the desired results.

I had to revert your removal of `entries.remove(&file.file_name());` because by just `continue`ing a user-defined desktop file is not able to override a global one. With it you can add `NoDisplay` to your user-defined file and neither will show up.

Reply:
This doesn’t work any more if `NoDisplay` is before `Exec` in the section.

Reply:
It should, since there's a `break` here? No other lines will be parsed.

Reply:
Ah right, sorry!